kahuna/public/js/components/gr-delete-image/gr-delete-image.js

```diff
@@ -0,0 +1,33 @@
+import angular from 'angular';
+import '../gr-confirm-delete/gr-confirm-delete';
+import './gr-delete-image.css!';
+
+export const imageDelete = angular.module('gr.imageDelete', ['gr.confirmDelete']);
+
+imageDelete.controller('grImageDeleteCtrl', ['$scope', 'mediaApi', function ($scope, mediaApi) {
+    var ctrl = this;
+
+    ctrl.delete = function () {
+        mediaApi.delete(ctrl.image)
+            .then((resp) => ctrl.onSuccess(resp))
+            .catch((err) => ctrl.onError(err));
```

Comment:
Note that if either of these callbacks are not provided this will error out. If they are required, could check for their presence when initialising the directive (fail early)?